pkg/issuer/acme/setup.go

```diff
@@ -167,6 +169,27 @@ func (a *Acme) Setup(ctx context.Context) error {
 	// registerAccount will also verify the account exists if it already
 	// exists.
 	account, err := a.registerAccount(ctx, cl)
+
+	// if we got an account successfully, we must check if the registered
+	// email is the same as in the issuer spec
+	registeredEmail := ""
+	if err == nil {
+		if len(account.Contact) > 0 {
+			registeredEmail = strings.Replace(account.Contact[0], "mailto:", "", 1)
+		}
+
+		// if they are different, we update the account
+		if registeredEmail != a.issuer.GetSpec().ACME.Email {
```

Comment:
Maybe an edge case, but just calling it out to be sure you've considered it... if a user does not specify an email address at all, I guess the `len(account.Contact)` will be 0, meaning `registeredEmail == ""`, meaning this whole thing should be skipped as expected. Fairly certain that is the case (especially now I've written it out here 😉), but just calling it out. Perhaps copy and paste some of this description into a comment? I know that future-us will appreciate it 😄

Reply:
Yep, since `""` is the zero-value of `spec.acme.email`.
I shall add a comment.